### PR TITLE
Create the archive bucket with the correct permissions and lifecycle config

### PR DESCRIPTION
**Pull request by djcp**
This creates the file archive bucket. We need to review the default policies to make sure they make sense.

### CODE REVIEW

File: lib/cluster/s3_distribution_bucket.rb [hunk 1]
```diff
@@ -74,7 +73,7 @@ def self.object_keys_to_delete(object)
       end
     end
 
-    def self.default_bucket_policy
+    def self.default_bucket_policy_for(bucket_name)
```
Comment (djcp) [inline lib/cluster/s3_distribution_bucket.rb:76]: @rute-santos this is probably where we'd want to change the bucket policy depending on whether or not it's for distribution or to serve as a file archive.

Comment (rute-santos) [inline lib/cluster/s3_distribution_bucket.rb:76]: Yes. How about if we have a S3BaseBucket class and S3DistributionBucket and S3ArchiveBucket subclasses where we specify the different policies?

Comment (djcp) [inline lib/cluster/s3_distribution_bucket.rb:76]: That's definitely an option. Can you pull together what this default bucket policy should look like for the archive bucket? I think we still don't want to put in the "forbid delete" rules for now.

**Comment by lbjay** [comment]
LGTM so far

**Pull request merged**